seqr/views/utils/permissions_utils.py

```diff
@@ -19,14 +22,49 @@ def get_project_and_check_permissions(project_guid, user, **kwargs):
     return project
 
 
+def project_has_anvil(project):
+    return anvil_enabled() and bool(project.workspace_namespace and project.workspace_name)
+
+
+def _anvil_to_seqr_permission(anvil_permission):
+    if anvil_permission['pending']:
+        return None
+    if anvil_permission['accessLevel'] in ['WRITER', 'OWNER']:
+        return CAN_EDIT
+    return CAN_VIEW
+
+
+def anvil_has_perm(user, permission_level, project):
+    if not project_has_anvil(project):
+        return False
+    workspace_acl = user_get_workspace_acl(user, project.workspace_namespace, project.workspace_name)
```

Comment:
do you really need to fetch the entire workspace access list to check if the currently authenticated user has access? It seems like since you are using the credentials of that user, there would be a faster way to check (for example, to check if the user is allowed to view the workspace, you could just do a head request against the workspace details and see if they get an access denier error, or something like that)

Reply:
Hmm... I re-visited the API list and found getting workspace details is the better one. The details of a workspace also have an `accessLevel` field which is what we need.

Reply:
okay but what if a user has view access to a workspace but does not have permission to view the workspace ACL?